### PR TITLE
Refactor Garden View page

### LINKED ISSUE #135: Refactor/redesign Garden View page
- [ ] The individual garden view page displays the card used on the Garden Management page. See #129. It should only need to be updated for one of these tickets.
- [ ] The individual garden view page has an old table  (`src/components/table.tsx`). It should use a new table (`src/components/Table`). Moreover, it should actually show the connected systems.

### PR DESCRIPTION
**Pull request by west270**
Closes #135

Refactored to improve the page and update which table component is being used.

I added permission checks that disable the form if user does not have `garden:update` permission.
I add websocket to update data when there is a garden update which should update the garden data when someone updates the connection information or does a garden sync.

### CODE REVIEW

File: src/components/GardenSyncButton.tsx [hunk 1]
```diff
@@ -36,16 +36,15 @@ const GardenSyncButton = ({
     execute({
       data: patchData,
     })
-      .then(() =>
+      .then(async () => {
```
Comment (jlrpnbbngtn) [inline src/components/GardenSyncButton.tsx:39]: It's not clear why the `async` keyword was added here and below.

File: src/pages/GardenAdminView/GardenAdminInfoCard.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import {
```
Comment (jlrpnbbngtn) [inline src/pages/GardenAdminView/GardenAdminInfoCard.tsx:1]: The filename should match the component name it's exporting. Whether that should be GardenAdminInfoCard or GardenInfoCard is up to you.

**Review by jlrpnbbngtn — CHANGES_REQUESTED** [review]

File: src/components/GardenConnectionForm/GardenConnectionForm.tsx [hunk 3]
```diff
@@ -51,18 +55,27 @@ const GardenConnectionForm = ({
         )}
       >
         <Form>
-          <Typography variant="h6">{title}</Typography>
-          {isCreateGarden ? <GardenName /> : null}
-          <ConnectionMethod />
-          <Divider sx={{ mt: 2, mb: 1 }} />
-          <Box sx={{ p: 1 }}>
-            <ConnectionHttpValues />
+          <fieldset disabled={!hasPermission('garden:update')}>
```
[the remark this answers is not in the document]
Comment (west270) [reply src/components/GardenConnectionForm/GardenConnectionForm.tsx:58]: I test the `FormControl` and it loses the border which could be added from styling but the form can still be edited when disabled which was why I was using the `fieldset`.

**Review by jlrpnbbngtn — APPROVED** [review]
Thank you, looks good!

**Pull request merged**